client-app/pages/category.vue

```diff
@@ -431,8 +420,14 @@ function resetFiltersWithKeyword() {
   }, FILTERS_RESET_TIMEOUT_IN_MS);
 }
 
+function applyInStockProductsFilter() {
+  filters.inStock = viewInStockProducts.value;
+  searchParams.value.filter = toFilterExpression(filters);
```

Comment:
`searchParams` is a computed property. 
There should be no value assignment here.

Reply:
This task may require a lot of refactoring